### PR TITLE
fix: make token expiry timezone agnostic

### PR DESCRIPTION
**Pull request by Kuro-Rui**
## Why this PR?
I'm currently using Python 3.11.3 and I found a TypeError while executing Client.get_me with an oauth token, and it comes from the check_token method, so here it is.

## Self-check

- [x] The changes are tested with [pre-commit](https://pre-commit.com/)
- [x] The changes pass unit testing with [pytest](https://pre-commit.com/)
- [x] The changes pass [mypy](http://mypy-lang.org/)

### CODE REVIEW

**Comment by NiceAesth** [comment]
Have been a bit busy, apologies for not responding to this PR yet. Will take a look at it later today

**Comment by NiceAesth** [comment]
Could you provide a stack trace / reproduction steps until then?

File: aiosu/v2/client.py [hunk 1]
```diff
@@ -122,8 +123,12 @@ def check_token(func: F) -> F:
     @functools.wraps(func)
     async def _check_token(self: Client, *args: Any, **kwargs: Any) -> Any:
         token = await self.get_current_token()
-        if datetime.utcnow() > token.expires_on:
-            await self._refresh()
+        try:
```
Comment (NiceAesth) [inline aiosu/v2/client.py:126]:
```suggestion
        if datetime.now(tz=timezone.utc) > token.expires_on:
```

Comment (NiceAesth) [reply aiosu/v2/client.py:126]: if this change is needed (not sure why you were getting typeerror from this), a try except is a poor way to handle it. just pass the timezone info always

**Review by NiceAesth — CHANGES_REQUESTED** [review]

File: aiosu/v2/client.py (hunk 1 above)
Comment (NiceAesth) [reply aiosu/v2/client.py:126]: i assume this is from passing a date object when creating the token. can just switch this function from a naive dt check

**Comment by NiceAesth** [comment]
ah, that is a bit annoying. would like reproduction steps to see if this is worth investigating further

**Comment by Kuro-Rui** [comment]
Sorry, been abandoning this for a while...
The new changes look great actually, I didn't think of using timestamp instead :p

**Comment by Kuro-Rui** [comment]
> ah, that is a bit annoying. would like reproduction steps to see if this is worth investigating further

I don't really know how to reproduce, but sometimes the datetime  object has a tz, so it returns a TypeError when checked with a datetime.utcnow() which has no tz.

**Review by NiceAesth — APPROVED** [review]
looks fine

**Pull request merged**